src/Servers/Kestrel/Core/src/Internal/Http3/Http3Connection.cs

```diff
@@ -341,23 +348,26 @@ private void UpdateStartingStreams(DateTimeOffset now)
                         stream.Abort(connectionError, (Http3ErrorCode)_errorCodeFeature.Error);
                     }
 
+                    lock (_sync)
+                    {
+                        OutboundControlStream?.Abort(connectionError, (Http3ErrorCode)_errorCodeFeature.Error);
```

Comment:
Is the lock really necessary? If we see a stale null OutboundControlStream, is it wrong to treat it as if it hadn't been fully created? I know this lock was there before and it doesn't really hurt if it's generally uncontested. I'm just curious.

Reply:
It's probably not necessary. Can look at cleaning it up in the future.